donkeycar/templates/train.py

```diff
@@ -1,1043 +1,191 @@
-                                  dtype=np.int64).values)
-        surgeon.add_job('delete_channels', model.get_layer(name),
-                        channels=channels)
-    # Delete channels
-    return surgeon.operate()
-
-
-def get_total_channels(model):
-    start = None
-    end = None
-    channels = 0
-    for layer in model.layers[start:end]:
-        if layer.__class__.__name__ == 'Conv2D':
-            channels += layer.filters
-    return channels
-
-
-def get_model_apoz(model, generator):
-    from kerassurgeon.identify import get_apoz
-    import pandas as pd
-
-    # Get APoZ
-    start = None
-    end = None
-    apoz = []
-    for layer in model.layers[start:end]:
-        if layer.__class__.__name__ == 'Conv2D':
-            print(layer.name)
-            apoz.extend([(layer.name, i, value) for (i, value)
-                         in enumerate(get_apoz(model, layer, generator))])
-
-    layer_name, index, apoz_value = zip(*apoz)
-    apoz_df = pd.DataFrame({'layer': layer_name, 'index': index,
-                            'apoz': apoz_value})
-    apoz_df = apoz_df.set_index('layer')
-    return apoz_df
-
-    
-def removeComments( dir_list ):
-    for i in reversed(range(len(dir_list))):
-        if dir_list[i].startswith("#"):
-            del dir_list[i]
-        elif len(dir_list[i]) == 0:
-            del dir_list[i]
-
-def preprocessFileList( filelist ):
-    dirs = []
-    if filelist is not None:
-        for afile in filelist:
-            with open(afile, "r") as f:
-                tmp_dirs = f.read().split('\n')
-                dirs.extend(tmp_dirs)
-
-    removeComments( dirs )
-    return dirs
-
-if __name__ == "__main__":
+    batch_size = cfg.BATCH_SIZE
+    dataset = TubDataset(tub_paths, test_size=(1. - cfg.TRAIN_TEST_SPLIT))
+    training_records, validation_records = dataset.train_test_split()
+    print('Records # Training %s' % (len(training_records)))
+    print('Records # Validation %s' % (len(validation_records)))
+
+    training = TubSequence(kl, cfg, training_records)
+    validation = TubSequence(kl, cfg, validation_records)
+
+    # Setup early stoppage callbacks
+    callbacks = [
+        EarlyStopping(monitor='val_loss', patience=cfg.EARLY_STOP_PATIENCE),
+        ModelCheckpoint(
+            filepath=output_path,
+            monitor='val_loss',
+            save_best_only=True,
+            verbose=1,
+        )
+    ]
+
+    kl.model.fit_generator(
+        generator=training,
+        steps_per_epoch=len(training),
+        callbacks=callbacks,
+        validation_data=validation,
+        validation_steps=len(validation),
+        epochs=cfg.MAX_EPOCHS,
+        verbose=cfg.VERBOSE_TRAIN,
+        workers=1,
+        use_multiprocessing=False
+    )
+
+
+def main():
     args = docopt(__doc__)
-    cfg = dk.load_config()
-    tub = args['--tub']
+    cfg = donkeycar.load_config()
+    tubs = args['--tubs']
     model = args['--model']
-    transfer = args['--transfer']
```

Comment:
Bringing back transfer should be easy, as there is basically no change compared to current behaviour.